src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,109 @@
-import { useState } from 'react';
+import React, { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
+interface Props {
+  onAdd: (movie: Movie) => void,
+
+}
+
+export const NewMovie: React.FC<Props> = ({
+  onAdd,
+}) => {
   // Increase the count after successful form submission
   // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+  const { 0: count, 1: setCount } = useState(0);
+  const { 0: hasSomeError, 1: setHasSomeError } = useState(false);
+
+  const { 0: title, 1: setTitle } = useState('');
```

Comment:
```suggestion
  const [title, setTitle] = useState('');
```
we usually write use states like this

Reply:
Array Assignment Pattern is 30% slower than Object Assignment Pattern. This happens due to the peculiarity of converting JS code into byte code when using the Array Assignment Pattern
https://tc39.es/ecma262/#prod-ArrayAssignmentPattern
https://tc39.es/ecma262/#prod-ObjectAssignmentPattern